client/blockbook/client.go

```diff
@@ -22,21 +20,66 @@ import (
 	"github.com/OpenBazaar/golang-socketio/protocol"
 	"github.com/OpenBazaar/multiwallet/client/transport"
 	"github.com/OpenBazaar/multiwallet/model"
+	"github.com/btcsuite/btcd/chaincfg"
 	"github.com/btcsuite/btcd/chaincfg/chainhash"
 	"github.com/btcsuite/btcutil"
+	"github.com/cpacia/bchutil"
 	"github.com/op/go-logging"
 	"golang.org/x/net/proxy"
 )
 
 var Log = logging.MustGetLogger("client")
 
+type wsWatchdog struct {
+	client    *BlockBookClient
+	done      chan struct{}
+	wsStopped chan struct{}
+}
+
+func newWebsocketWatchdog(client *BlockBookClient) *wsWatchdog {
+	return &wsWatchdog{
+		client:    client,
+		done:      make(chan struct{}, 0),
+		wsStopped: make(chan struct{}, 5),
+	}
+}
+
+func (w *wsWatchdog) guardWebsocket() {
+	for {
+		select {
+		case <-w.wsStopped:
+			time.Sleep(1 * time.Second)
+			Log.Warningf("reconnecting websocket...")
+			w.client.SocketClient.Close()
+			w.client.SocketClient = nil
+			for len(w.wsStopped) > 0 {
```

Comment:
This will require that `bark` is called twice before a reconnect can happen. I'd do something like this:
```
for {
  select {
    case <-w.wsStopped:
    default:
    break
  }
}
```